app-frontend/src/app/components/scenes/sceneDetailModal/sceneDetailModal.controller.js

```diff
@@ -58,4 +63,90 @@ export default class SceneDetailModalController {
     closeWithData(data) {
         this.close({$value: data});
     }
+
+    toggleMetadataEdit() {
+        this.isEditMetadata = !this.isEditMetadata;
+        if (!this.isEditMetadata) {
+            this.updateMetadata();
+        }
+    }
+
+    setAqcDateDisplay() {
+        return this.scene.filterFields && this.scene.filterFields.acquisitionDate ?
+            this.formatAcqDate(this.scene.filterFields.acquisitionDate) :
+            'MM/DD/YYYY';
+    }
+
+    updateMetadata() {
+        // TODO: visibility and data source should be editable eventually
+        // TODO: sunAzimuth and sunElevation can't be changed, might be backend issue?
+        this.isUploadDone = false;
+        if (!this.newFilterFields.acquisitionDate) {
+            this.newFilterFields.acquisitionDate = this.scene.filterFields.acquisitionDate;
+        }
+        this.scene = Object.assign(this.scene, {
+            'modifiedAt': this.Moment().toISOString(),
+            'modifiedBy': this.scene.owner,
+            'sceneMetadata': this.newSceneMetadata,
+            'filterFields': this.newFilterFields
+        });
+        this.sceneService.update(this.scene).then(
+            () => {
+                this.isUploadDone = true;
+            },
+            () => {
+                this.isUploadDone = false;
+            }
+        );
+    }
+
+    formatAcqDate(date) {
+        return date.length ? this.Moment(date).format('MM/DD/YYYY') : 'MM/DD/YYYY';
+    }
+
+    openDatePickerModal(date) {
+        if (this.activeModal) {
+            this.activeModal.dismiss();
+        }
+
+        this.activeModal = this.$uibModal.open({
+            component: 'rfDatePickerModal',
+            windowClass: 'auto-width-modal',
+            resolve: {
+                config: () => Object({
+                    selectedDay: this.Moment(date)
+                })
+            }
+        });
+
+        this.activeModal.result.then(
+            selectedDay => {
+                this.updateAcquisitionDate(selectedDay);
+            });
+    }
+
+    updateAcquisitionDate(selectedDay) {
+        this.newFilterFields.acquisitionDate = selectedDay.toISOString();
```

Comment:
If the user cancels the date picker without selecting a date, `selectedDay` is `false`. This causes this line to throw an error when it assumes `toISOString()` is a method on `selectedDay`. It would be good to wrap the method body in an `if (selectedDay)`.